be2-scala/src/test/scala/ch/epfl/pop/decentralized/ConnectionMediatorSuite.scala

```diff
@@ -0,0 +1,79 @@
+package ch.epfl.pop.decentralized
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.testkit.{TestKit, TestProbe}
+import ch.epfl.pop.model.network.method.Heartbeat
+import ch.epfl.pop.pubsub.ClientActor.ClientActorMessage
+import ch.epfl.pop.pubsub.MessageRegistry
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.funsuite.{AnyFunSuiteLike => FunSuiteLike}
+import org.scalatest.matchers.should.Matchers
+
+class ConnectionMediatorSuite extends TestKit(ActorSystem("ConnectionMediatorSuiteActorSystem")) with FunSuiteLike with Matchers with BeforeAndAfterAll {
+  override def afterAll(): Unit = {
+    // Stops the test actor system
+    TestKit.shutdownActorSystem(system)
+  }
+
+  test("ConnectionMediator tells monitor if some or no server are connected") {
+    val mockMonitor = TestProbe()
+    val server = TestProbe()
+
+    val connectionMediatorRef = system.actorOf(
+      ConnectionMediator.props(mockMonitor.ref, ActorRef.noSender, ActorRef.noSender, MessageRegistry())
+    )
+
+    // Register server
+    connectionMediatorRef ! ConnectionMediator.NewServerConnected(server.ref)
+
+    mockMonitor.expectMsg(Monitor.AtLeastOneServerConnected)
```

Comment:
It may be helpful to use our own timeout duration here too (also for `xyz..expectNoMessage()`.

Reply:
done !